xbmc/utils/URIUtils.cpp

```diff
@@ -517,32 +509,57 @@ bool URIUtils::IsProtocol(const std::string& url, const std::string &type)
   return StringUtils::StartsWithNoCase(url, type + "://");
 }
 
-bool URIUtils::PathStarts(const std::string& url, const char *start)
+bool URIUtils::PathHasParent(const std::string& path, const std::string& parent, bool translate /* = false */)
```

Comment:
Just make `path` and `parent` values instead of const references because you need to copy them into internal strings anyway.